Linux/build_openssl.sh

```diff
@@ -81,7 +81,7 @@ fi
 # Mitigation flags
 MITIGATION_OPT=""
 MITIGATION_FLAGS=""
-CC_VERSION=`gcc -dumpversion`
+CC_VERSION=`gcc -dumpversion | cut -d"." -f1`
```

Comment:
It needs to align to the SGX SDK code:
https://github.com/intel/linux-sgx/blob/2ee53db4e8fd25437a817612d3bcb94b66a28373/buildenv.mk#L227

Reply:
Please note that the current implementation is broken.

https://github.com/intel/intel-sgx-ssl/blob/57cb0c763772e00fbc32781673fde2d7d762827b/Linux/build_openssl.sh#L102-L104

Assuming a GCC version of `10.3.0` given the following script:

```bash
#!/usr/bin/env bash
if [[ 10.3.0 -ge 8 ]]; then
  echo "GCC at least version 8"
fi
```

Execution of the script results in the following error:

```
./test.sh: line 2: [[: 10.3.0: syntax error: invalid arithmetic operator (error token is ".3.0")
```

This makes sense as the version string `10.3.0` isn't a valid number.

To fix this, the latest version of this PR extracts the GCC major version similar to `linux-sgx` here: https://github.com/intel/linux-sgx/blob/2ee53db4e8fd25437a817612d3bcb94b66a28373/buildenv.mk#L91

Reply:
That means 
`CC_NO_LESS_THAN_8 := $(shell expr $(CC_VERSION) \>\= "8")` at
https://github.com/intel/linux-sgx/blob/2ee53db4e8fd25437a817612d3bcb94b66a28373/buildenv.mk#L227
will lead to the syntax error when GCC version of `10.3.0`, right?

Reply:
OK. I see the difference between the behavior of shell script and makefile. Both the SGX code and your suggestion are correct.